### PR TITLE
[FixaMinGata] Adds a cobrand hook (threshold for responsiveness top 5 list)

### LINKED ISSUE #1957: Changed "Top 5 responsive councils" criteria
The "Top 5 responsive councils" list on FixaMinGata shows a municipality that has only solved one problem (back in 2013) at the top, since that action took only about four minutes, and that beats any other average response time on the site.

stats1

This does not seem like a fair representation of responsiveness, although I'm not sure what a better criteria would be. Perhaps there should be a minimum number of reports to be able to make the list. Perhaps reports of a certain age should not be taken into account.

### PR DESCRIPTION
**Pull request by joesiltberg**
This adds an optional threshold so that a body with very few fixed reports don't end up in the top 5 list under /reports.

I haven't tested the change in App/Controller/Dashboard.pm, am I supposed to be able to see the responsiveness somewhere under /dashboard? Or should I look somewhere else?

Fixes #1957

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Dashboard.pm [hunk 1]
```diff
@@ -292,7 +292,7 @@ sub generate_summary_figures {
 sub generate_body_response_time : Private {
     my ( $self, $c ) = @_;
 
-    my $avg = $c->stash->{body}->calculate_average;
+    my $avg = $c->stash->{body}->calculate_average($c->$cobrand->call_hook("body_responsiveness_threshold") || 0);
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Dashboard.pm]: `s/$cobrand/cobrand` here should fix the tests.

File: perllib/FixMyStreet/DB/Result/Body.pm [hunk 2]
```diff
@@ -204,7 +204,7 @@ sub get_cobrand_handler {
 }
 
 sub calculate_average {
-    my ($self) = @_;
+    my ($self, $threshold) = @_;
```
Comment (dracos) [inline perllib/FixMyStreet/DB/Result/Body.pm]: Perhaps here have a safety `$threshold ||= 0;` line, and then you don't need to make sure it's 0 when calling the function in the two places.

**Comment by joesiltberg** [comment]
Thanks for the feedback @dracos ! Should I make the change for filtering out hidden reports in the regular code (i.e. not a cobrand hook)?

**Comment by dracos** [comment]
Yes, I can't see why it shouldn't be in the main code, thanks!

**Comment by dracos** [comment]
I've reworded one commit (wasn't FixaMinGata specific :) ), and put the fixes from the second commit in the first one (in general, if you say "also" in a commit message, it should probably be a separate commit ;) Makes it easier to squash/fixup). Thanks, will merge when tests pass.

**Comment by joesiltberg** [comment]
Great, thanks!

**Pull request merged**